server/sonar-web/src/main/js/helpers/testUtils.ts

```diff
@@ -56,11 +56,18 @@ export function keydown(keyCode: number): void {
 }
 
 export function elementKeydown(element: ShallowWrapper, keyCode: number): void {
-  element.simulate('keydown', {
+  const event = {
     currentTarget: { element },
     keyCode,
     preventDefault() {}
-  });
+  };
+
+  if (typeof element.type() === 'string') {
+    // `type()` is string for native dom elements
+    element.simulate('keydown', event);
+  } else {
+    element.prop<Function>('onKeyDown')(event);
```

Comment:
Well, everywhere it's used the pattern is the same:
```javascript
elementKeydown(wrapper.find('SearchBox'), 40);
wrapper.update();
```
so it doesn't make sense to update `element`, since it will update the `SearchBox` itself and not its parent.

Reply:
Ah yeah, true